### ISSUE TITLE
7.0.3 regression: exception while parsing source with postcss-safe-parser

### ISSUE THREAD

**Issue by AviVahl** [postcss/postcss#1192]
We are using `postcss` alongside [postcss-safe-parser](https://github.com/postcss/postcss-safe-parser) in [stylable](https://github.com/wix/stylable), and use that infrastructure in [stylable-intelligence](https://github.com/wix/stylable-intelligence)

since the recently released postcss@7.0.3, I see the following crash in multiple tests of stylable-intelligence:
```
TypeError: Cannot read property '0' of undefined
      at SafeParser.colon (node_modules/postcss/lib/parser.es6:460:17)
      at SafeParser.colon [as precheckMissedSemicolon] (node_modules/postcss-safe-parser/lib/safe-parser.es6:53:24)
      at SafeParser.precheckMissedSemicolon [as decl] (node_modules/postcss/lib/parser.es6:206:10)
      at SafeParser.decl (node_modules/postcss-safe-parser/lib/safe-parser.es6:32:7)
      at SafeParser.decl [as other] (node_modules/postcss/lib/parser.es6:140:12)
      at SafeParser.other (node_modules/postcss/lib/parser.es6:56:16)
      at parse (node_modules/postcss-safe-parser/lib/safe-parse.es6:9:10)
      at Object.safeParse (node_modules/@stylable/core/src/parser.ts:15:12)
      at /home/avi/projects/stylable-intelligence/node_modules/@stylable/core/src/create-infra-structure.ts:34:28
      at processContent (node_modules/@stylable/core/src/cached-process-file.ts:51:12)
      at Object.process (node_modules/@stylable/core/src/cached-process-file.ts:41:27)
      at Stylable.process (node_modules/@stylable/core/src/stylable.ts:112:35)
      at files.forEach.file (src/lib/provider.ts:807:38)
      at Array.forEach (<anonymous>)
      at newFindRefs (src/lib/provider.ts:806:15)
      at Object.getRefs (src/lib/provider.ts:917:12)
      at Object.getReferences (test-kit/asserters.ts:63:12)
      at Context.<anonymous> (test/lib/references.spec.ts:61:26)
```
(the above is caught with source-map-support, so source locations are provided)
v7.0.2 worked properly

**Comment by AviVahl** [comment]
probably was caused by https://github.com/postcss/postcss/pull/1186/files#diff-652c1c1ab9703a9fb1d789af4262c518

**Comment by ai** [comment]
Thanks. I will look it in next 15 minutes

**Comment by ai** [comment]
Can your give me some CSS to reproduce error?

**Comment by AviVahl** [comment]
```
:vars {
    jarjar: binks;
    varvar: pink;
}

.local {
    color: value(jarjar) value(varvar);
}
```

**Comment by ai** [comment]
@janczer I will fix this issue, but can you review rest of your PR for similar issues. It is OK to make issues (I made many mistakes too), but we must to learn from them.

**Comment by ai** [comment]
@AviVahl @janczer fixed 20f80a8

**Comment by AviVahl** [comment]
Verified as fixed from our end. Thanks for the updated release!
